array.c

```diff
@@ -2551,18 +2551,54 @@ rb_ary_aset(int argc, VALUE *argv, VALUE ary)
 
 /*
  *  call-seq:
- *     ary.insert(index, obj...)  -> ary
+ *    array.insert(index, *objects) -> self
  *
- *  Inserts the given values before the element with the given +index+.
+ *  Inserts given +objects+ before or after the element at +offset+ index;
+ *  returns +self+.
+ *
+ *  Argument +index+ must be an
+ *  {Integer-convertible object}[doc/implicit_conversion_rdoc.html#label-Integer-Convertible+Objects].
+ *
+ *  ---
+ *
+ *  When +index+ is non-negative, inserts all given +objects+
+ *  before the element at offset +index+:
+ *    a = [:foo, 'bar', baz = 2]
```

Comment:
In your [previous PR you did remove](https://github.com/ruby/ruby/pull/3204/files#diff-24411fb2a634ee46e29925b04767d15eL2006) the `bar = 2`, which I consider a good change because it had an unneeded complexity. But is introduced again here.

IMO and for consistency sake I'd recommend to don't have an array element with an assignment. 
```suggestion
 *    a = [:foo, 'bar', 2]
```

Reply:
Thanks, @esparta.  Meant to do that.

Reply:
no problem, glad to help even this minimal.